### ISSUE TITLE
VsVim for Mac: exiting insert mode requires multiple presses of Esc 

### ISSUE THREAD

**Issue by coolbluewater** [VsVim/VsVim#2766]
On my machine a reliable repro is this C# function call:

```
using System;
class Foo
{
    public static void Bar()
    {
        var x = "Hello";
        var y = "There";
        Console.WriteLine("{0}, {1}, {2}", Math.PI, x, y);
    }
}
```
Note:  This won't repro on non-working code, so make sure there are no squigglies in the above code before proceeding.

Position the cursor at the comma after `x` and type `i, y` to insert a comma, space, and a `y`.

Now hit Esc to return to normal mode.

Expect: To hit Esc just once.
Actual: Need to hit Esc three times before the cursor turns to a block.

@nosami

**Comment by nosami** [comment]
Can confirm. Surprised I haven't noticed this myself.... doesn't seem like VsVim is handling any of the first 2 escape keypresses.

Should be an easy fix though.

**Comment by nosami** [comment]
<img width="1081" alt="image" src="https://user-images.githubusercontent.com/667194/74090876-9f8cfe00-4aa8-11ea-9b53-9d5d62539f8f.png">

Before pressing any escape key, both the tooltip and the IntelliSense windows are visible at this point. The first Esc closed the IntelliSense window, the 2nd one closes the tooltip and the 3rd escape keypress gets handled by VsVim and things return to normal.

**Comment by nosami** [comment]
Any QuickInfo window (tooltip) that is being displayed should also probably be hidden on `Esc` press.

**Comment by nosami** [comment]
@coolbluewater 

I just tested this out on Windows to compare. It behaves how you want in VS2019, so I'll make sure it works the same on Mac.

**Comment by nosami** [comment]
Fixed in 2.8.0.8

**Comment by theorlangur** [comment]
I have a similar issue on Visual Studio 2019 and a fix with 'Esc' works fine, but in my case I've re-mapped esc to 'jk' and in this case whatever drop downs, tooltip hints are displayed are not closed when leaving an insert mode. 
Could this maybe also be addressed?
